sim/paladin/avengers_shield.go

```diff
@@ -0,0 +1,59 @@
+package paladin
+
+import (
+	"github.com/wowsims/sod/sim/core"
+	"github.com/wowsims/sod/sim/core/proto"
+	"time"
+)
+
+func (paladin *Paladin) registerAvengersShield() {
+	if !paladin.hasRune(proto.PaladinRune_RuneLegsAvengersShield) {
+		return
+	}
+
+	// Avenger's Shield hits up to 3 targets.
+	results := make([]*core.SpellResult, min(3, paladin.Env.GetNumTargets()))
+
+	paladin.GetOrRegisterSpell(core.SpellConfig{
+		ActionID:    core.ActionID{SpellID: 407669},
+		SpellCode:   SpellCode_PaladinAvengersShield,
+		SpellSchool: core.SpellSchoolHoly,
+		DefenseType: core.DefenseTypeMelee,
+		ProcMask:    core.ProcMaskMeleeMHSpecial,
```

Comment:
Is avenger's shield considered a melee ability?

Reply:
Ah its a spell...but it doesn't get resisted...and now you have opened my eyes to possible inconsistencies in multiple pally abilities... :sweat_smile:

Reply:
Added flags to get it to not miss, not resist, spell crit _chance_, melee crit _multiplier_.